src/main/java/io/debezium/connector/cassandra/SchemaHolder.java

```diff
@@ -5,154 +5,292 @@
-                return null;
+        @Override
+        public void onTableChanged(final TableMetadata newTableMetadata, final TableMetadata oldTableMetaData) {
+            if (newTableMetadata.getOptions().isCDC()) {
+                // if it was cdc before and now it is too, add it, because its schema might change
+                // however if it is CDC-enabled but it was not, update it in schema too because its cdc flag has changed
+                // this basically means we add / update every time if new has cdc flag equals to true
+                addOrUpdateTableSchema(new KeyspaceTable(newTableMetadata),
+                        new KeyValueSchema(kafkaTopicPrefix, newTableMetadata, sourceInfoStructMaker));
+            }
+            else if (oldTableMetaData.getOptions().isCDC()) {
+                // if new table is not on cdc anymore, and we see the old one was, remove it
+                removeTableSchema(new KeyspaceTable(newTableMetadata));
             }
-            return SchemaBuilder.struct().name(NAMESPACE + "." + getValueName(kafkaTopicPrefix, tm))
-                    .field(Record.TIMESTAMP, Schema.INT64_SCHEMA)
-                    .field(Record.OPERATION, Schema.STRING_SCHEMA)
-                    .field(Record.SOURCE, sourceInfoStructMaker.schema())
-                    .field(Record.AFTER, RowData.rowSchema(tm))
-                    .build();
+            try {
+                LOGGER.debug("Detected alternation in schema of {}.{} (previous cdc = {}, current cdc = {})",
+                        newTableMetadata.getKeyspace().getName(),
+                        newTableMetadata.getName(),
+                        oldTableMetaData.getOptions().isCDC(),
+                        newTableMetadata.getOptions().isCDC());
+                // else if it was not cdc before nor now, do nothing with schema holder
+                // but add it to Cassandra for subsequent deserialization path in every case
+                final CFMetaData rawNewMetadata = CFMetaData.compile(newTableMetadata.asCQLQuery(),
+                        newTableMetadata.getKeyspace().getName());
+                final CFMetaData oldMetadata = Schema.instance.getCFMetaData(oldTableMetaData.getKeyspace().getName(), oldTableMetaData.getName());
+                // we need to copy because CFMetaData.compile will generate new cfId which would not match id of old metadata
+                final CFMetaData newMetadata = rawNewMetadata.copy(oldMetadata.cfId);
+                oldMetadata.apply(newMetadata);
+                LOGGER.info("Updated table [{}] in Schema Instance.", newTableMetadata.asCQLQuery());
+            }
+            catch (Exception e) {
+                LOGGER.warn("Error happened while reacting on changed table {}.{} in Schema instance", newTableMetadata.getKeyspace(), newTableMetadata.getName(), e);
+            }
+        }
+
+        @Override
+        public void onUserTypeAdded(final UserType type) {
+        }
```

Comment:
Why all the empty implementations?

Reply:
> Why all the empty implementations?

From my understanding, this is a required method to be overriden to implement SchemaChangeListener in Cassandra, but Cassandra Connector doesn't actually need to do anything for the case onUserTypeAdded so the function is listed here but kept empty.

Reply:
Yes, makes sense. Thx! I'd suggest to put comments in there:

    // not relevant to Debezium

Reply:
> Yes, makes sense. Thx! I'd suggest to put comments in there:
> 
> ```
> // not relevant to Debezium
> ```

Added comments accordingly.